[Security] 9.4.1 release notes (#6380)
elastic/docs-content: release-notes/elastic-security/index.md

% *

## 9.4.1 [elastic-security-9.4.1-release-notes]

### Fixes [elastic-security-9.4.1-fixes]

> * Fixes an issue where detection rules failed with errors when the executing user lacked read access to entity store indices. Entity store enrichment is now silently skipped for users without entity store index access [#267787]({{kib-pull}}267787).

Reviewer: ```suggestion
* Fixes an issue where detection rules failed with errors when their executing users lacked read access to entity store indices. Entity store enrichment is now silently skipped for users without entity store index access [#267787]({{kib-pull}}267787).
```

* Fixes an issue where the Security alert flyout failed to render its header, content, and footer when Discover was embedded in a dashboard [#267321]({{kib-pull}}267321).

> * Fixes the events histogram and host/network event count panels on the **Overview** dashboard, which incorrectly included alert documents in event counts, and ensures compatibility with cross-cluster search so events from remote clusters are correctly included [#265561]({{kib-pull}}265561).

Reviewer 2: @ashokaditya does https://github.com/elastic/kibana/pull/268216 need to be included in the 9.4.1 Security release notes? If yes, can you review the summary I added below for accuracy and completeness?

@romulets two questions about https://github.com/elastic/kibana/pull/268170:
- Does this PR also need to be included in the 9.4.1 Security release notes? If yes, can you review the summary I added below for accuracy and completeness?
- Was a doc issue filed to document the `maxTimeWindowSize` parameter that 268170 introduces? If not, we'll need one please and thank you! 
 
```suggestion
* Fixes the events histogram and host/network event count panels on the **Overview** dashboard, which incorrectly included alert documents in event counts, and ensures compatibility with cross-cluster search so events from remote clusters are correctly included [#265561]({{kib-pull}}265561).
* Fixes an incorrect ordering of the **Launchpad** item in the classic global side navigation. The item now appears above **Manage**, consistent with {{elastic-sec}} own classic navigation [#268216]({{kib-pull}}268216).
* Fixes a performance degradation in Entity Store log extraction where, in lagging environments, the extraction window grew unboundedly, causing each subsequent probe to sort an ever-larger set of documents in {{esql}}, leading to a slowdown death spiral. Also introduces a configurable `maxTimeWindowSize` parameter (default: `15m`) that limits the probe window per extraction run, keeping query cost bounded regardless of how far behind the engine is [#268170]({{kib-pull}}268170).
```

Reviewer 3: Thanks @nastasha-solomon. Yes it should be included and the note is spot on.

Reviewer 4: Thanks @nastasha-solomon, will answer behalf of @romulets.
There is an additional parameter that was introduced as part of this PR (`excludedIndexPatterns`). tried to capture all the changes in the following paragraph, let me know what you think.

_Fixes an issue where the Entity Store log extraction window could grow unboundedly in lagging environments, causing ES|QL probe cost to spiral. Adds a `maxTimeWindowSize` parameter (default: `15m`) to the install and update APIs to cap each probe's window size. Also adds an `excludedIndexPatterns` parameter that lets you exclude specific index patterns from log extraction.
By default, the Entity Store extracts entities from all data sources defined in the Default Security data view; use excludedIndexPatterns to skip patterns you don't want to include[#268170]({{kib-pull}}268170)._

Reviewer 4: Here is the doc [ticket](https://github.com/elastic/docs-content/issues/6398), let me know if anything else should be added.

Author: Hey @nastasha-solomon and  @uri-weisman, it looks like these new parameters are just API parameters that have already been added to the API reference in https://github.com/elastic/kibana/pull/268170/changes? Please let me know if I'm missing something and if these should also be doc'ed elsewhere in the docs. Thanks!

Author: Actually, it seems like we may want to document these in the [Troubleshooting](https://www.elastic.co/docs/solutions/security/advanced-entity-analytics/entity-store#entity-store-troubleshoot) section since this describes the parameters for tuning log extraction behavior. @uri-weisman WDYT?

Reviewer 4: I agree, adding it to the troubleshooting section can be very useful for customers.

Author: Sounds good! Since the parameters have already been added to the OAS spec in the dev PR, I'll repurpose the ticket you opened to track the narrative docs update instead. 👍

Reviewer 4: Thanks @natasha-moore-elastic !

> * Fixes the events histogram and host/network event count panels on the **Overview** dashboard, which incorrectly included security alert documents in event counts, and ensures Cross-Project Search ({{cps-init}}) compatibility so events from remote-cluster projects are correctly included [#265561]({{kib-pull}}265561).

Reviewer 2: @NicholasPeretti this PR affects serverless only, right?

```suggestion
```

Reviewer 5: No, it also affects stateful installations that use CCS (which is the stateful version of CPS) 😄

Reviewer 2: Gotcha - can you reframe the summary so it reflects the fixes made for CCS deployments?

Reviewer 5: How about this?

```suggestion
* Fixes the events histogram and host/network event count panels on the **Overview** dashboard, which incorrectly included security alert documents in event counts, and ensures compatibility with Cross-Cluster Search and Cross-Project Search ({{cps-init}}) so events from remote clusters are correctly included [#265561]({{kib-pull}}265561).
```

Reviewer 2: Just a small tweak to remove CPS since these release notes are for Stack only

```suggestion
* Fixes the events histogram and host/network event count panels on the **Overview** dashboard, which incorrectly included alert documents in event counts, and ensures compatibility with cross-cluster search so events from remote clusters are correctly included [#265561]({{kib-pull}}265561).
```
